python/mrd_2_xnat.py

```diff
@@ -0,0 +1,293 @@
+    ]
+
+    return xnat_mrd_list
+
+
+def handle_sequence_params(xnat_mrd_list: list[list[Any]]) -> list[list[Any]]:
+    """
+    Filter out unwanted sequence parameters from xnat_mrd_list.
+
+    This function removes diffusion parameters entirely and keeps only the first
+    entry (index 0) for specific single-value sequence parameters.
+    """
+    # Remove diffusion parameters entirely
+    xnat_mrd_list = [
+        elem
+        for elem in xnat_mrd_list
+        if elem[:2] != ["sequenceParameters", "diffusion"]
+    ]
+
+    # Keep only first entry (index 0) for single-value parameters
+    single_entries = ["TR", "TE", "flipAngle_deg", "echo_spacing", "TI"]
+    for entry in single_entries:
+        xnat_mrd_list = [
+            elem
+            for elem in xnat_mrd_list
+            if elem[:2] != ["sequenceParameters", entry] or elem[2] == 0
+        ]
+
+    return xnat_mrd_list
+
+
+def handle_meas_info(xnat_mrd_list: list[list[Any]]) -> list[list[Any]]:
+    """
+    Filter out unwanted measurement info from xnat_mrd_list.
+
+    This function removes redundant measurement information entries and keeps only the
+    first entry (index 0) for parameters that typically have multiple instances but
+    where only the first instance is needed.
+
+    """
+    xnat_mrd_list = [
+        elem
+        for elem in xnat_mrd_list
+        if elem[:2] != ["measurementInformation", "measurementDependency"]
+        or elem[2] == 0
+    ]
+    xnat_mrd_list = [
+        elem
+        for elem in xnat_mrd_list
+        if elem[:2]
+        != [
+            "measurementInformation",
+            "referencedImageSequence",
+            "referencedSOPInstanceUID",
+        ]
+        or elem[3] == 0
+    ]
+
+    return xnat_mrd_list
+
+
+def create_final_xnat_mrd_dict(
+    xnat_mrd_list: list[list[Any]],
+    ismrmrd_dict: dict[str, Any],
+    xnat_mrd_dict: dict[str, str],
+) -> dict[str, Any]:
+    """
+    Create final xnat_mrd_dict by building parameter paths and extracting values.
+
+    Converts parameter paths to XNAT-compatible key strings and retrieves corresponding
+    values from the ismrmrd_dict. Handles XNAT field length limitations.
+    """
+    for param_path in xnat_mrd_list:
+        # Build XNAT key path
+        key_parts = ["mrd:mrdScanData"]
+        key_parts.extend(str(part) for part in param_path if isinstance(part, str))
+        ckey = "/".join(key_parts)
+
+        # This field seems to be too long for xnat
+        if "parallelImaging/accelerationFactor/kspace_encoding_step" in ckey:
+            ckey = ckey.replace("kspace_encoding_step", "kspace_enc_step")
+        xnat_mrd_dict[ckey] = get_dict_values(ismrmrd_dict, param_path)
+
+    xnat_mrd_dict["mrd:mrdScanData/acquisitionSystemInformation/coilLabelList"] = "TEMP"
+
+    return xnat_mrd_dict
+
+
+def check_header_valid_convert_to_dict(
+    xml_scheme_filename: Path, ismrmrd_header: bytes
```

Comment:
```suggestion
    xml_schema_filename: Path, ismrmrd_header: bytes
```
It's still as 'scheme' here rather than 'schema'